lib/plugin-loader.js

```diff
@@ -959,12 +971,47 @@ PluginLoader.prototype = {
               bootstrapLogger.severe(e.message);
               resolve(); // We don't want to reject here. Error gets caught down stream
             });
+          })
+        }).then(() => { /* Obtains and stores the endpoints exposed by the agent */
+          requestOptions.path = '/application/info';
+          if(config.node.mediationLayer &&
+            config.node.mediationLayer.enabled &&
+            config.node.mediationLayer.server){
+              requestOptions.host = config.node.mediationLayer.server.hostname
+              requestOptions.port = config.node.mediationLayer.server.gatewayPort
+            }
+          return new Promise((resolve, reject) => { 
+            httpApi.get(requestOptions, (res) => {
+              const { statusCode } = res; // TODO: Check status code for bad status
```

Comment:
considering the gateway often starts slower than app-server, i think you need some sort of loop w/ timeout here because it's highly likely to fail on the first request.